MonoGame.Framework/Vector3.cs

```diff
@@ -522,7 +530,7 @@ public bool Equals(Vector3 other)
         /// <returns>Hash code of this <see cref="Vector3"/>.</returns>
         public override int GetHashCode()
         {
-            return (int)(this.X + this.Y + this.Z);
+            return X.GetHashCode() + Y.GetHashCode() + Z.GetHashCode();
```

Comment:
I don't know XNA generates hashcodes for Vectors, and no-one should know until MS release the source for it.  But simply adding the hashcodes for the separate components doesn't make for a good hashcode.  Granted, the existing version did that anyway, and it truncated floats, so it wasn't the best, but your new version suffers the same problem.  Take the following three vectors.  They will generate the same hashcode because the component hashcodes are simply added together.

{ 1.0f, 2.0f, 3.0f }
{ 3.0f, 1.0f, 2.0f }
{ 2.0f, 3.0f, 1.0f }

The component hashcodes ideally should be combined in a manner that is not commutative (plus is commutative because it doesn't matter what order the operands are in).